geode-core/src/main/java/org/apache/geode/distributed/internal/locks/DLockGrantor.java

```diff
@@ -535,6 +551,26 @@ void handleLockBatch(DLockRequestMessage request) throws InterruptedException {
     }
   }
 
+  void recordMemberDepartedTime(InternalDistributedMember owner) {
+    synchronized (membersDepartedTime) {
```

Comment:
change this code to only call System.currentTimeMillis once.
How about using a LinkedHashMap? I think you could then get rid of the membersDepartedTimeKeptSize. The order of iteration should be from the first one added to the last. So that would always be the oldest to the most recent. As soon as you find and entry that is not expired you can quit the iteration.